### PR TITLE
Rewrite engine to use SDL only [WIP]

### PR DESCRIPTION
**Pull request by nerzhul**

### CODE REVIEW

File: source/Irrlicht/Irrlicht.cpp [hunk 1]
```diff
@@ -64,53 +64,7 @@ namespace irr
...
-		if (dev && !dev->getVideoDriver() && params.DriverType != video::EDT_NULL)
-		{
-			dev->closeDevice(); // destroy window
-			dev->run(); // consume quit message
-			dev->drop();
-			dev = 0;
-		}
-
-		return dev;
+		return new CIrrDeviceLinux(params);
```
Comment (sfan5) [inline source/Irrlicht/Irrlicht.cpp:67]: This seems to mix the (old) Linux/X11 device with all new SDL code.

I have doubt that this works on Windows or macOS.

Comment (nerzhul) [reply source/Irrlicht/Irrlicht.cpp:67]: it's not ready yet, just here for the posterity as a startup :)

Comment (sfan5) [reply source/Irrlicht/Irrlicht.cpp:67]: What I'm trying to say is: it might be better to create a new CIrrDeviceSDL2 then

Comment (nerzhul) [reply source/Irrlicht/Irrlicht.cpp:67]: yes it should be a good option, i will try to take more time to make it work properly :)

Comment (nerzhul) [reply source/Irrlicht/Irrlicht.cpp]: it's done now, linux device is dropped. I need to rebind events. hopfully there is a SDL1 backend to help us to translate some orders from WM system, but not all :)

**Comment by nerzhul** [comment]
works a bit, keymap is currently broken and mouse is not show on screen. Didn't tested android yet, sounds to be handled by SDL itself and can be improved in irrlicht internals directly

**Pull request closed without merging**